crates/polars-ops/src/series/ops/replace.rs

```diff
@@ -33,7 +33,20 @@ pub fn replace(
         return Ok(default);
     }
 
-    let old = old.strict_cast(s.dtype())?;
+    let old = match (s.dtype(), old.dtype()) {
+        (s_dt, old_dt) if s_dt == old_dt => old.clone(),
+        #[cfg(feature = "dtype-categorical")]
+        (DataType::Categorical(opt_rev_map, ord), DataType::Utf8) => {
+            let dt = opt_rev_map
```

Comment:
I read this as a `clone_if_enum`. Do we have this pattern more often? Might make a dedicate method for that.

Reply:
For this PR, I would like to leave this as. But in the future I would like to tackle the cast to Categorical which is compatible instead of first casting to Categorical(None) and then remapping